big-o-notation/big-o.js

```diff
@@ -0,0 +1,44 @@
+/* exported uniqueQuadratic, uniqueLinear */
+
+function uniqueLinear(words) {
+  const seen = {};                  // 1 * 1 = O(1)
+  const unique = [];                // 1 * 1 = O(1)
+  for (
+    let i = 0;                      // 1 * 1 = O(1)
+    i < words.length;               // 2 * n = O(n)
+    i++                             // 2 * n = O(n)
+  ) {
+    const word = words[i];          // 2 * n = O(n)
+    if (!seen[word]) {              // 2 * n = O(n)
+      seen[word] = true;            // 2 * n = O(n)
+      unique[unique.length] = word; // 3 * n = O(n)
+    }
+  }
+  return unique;                    // 1 * 1 = O(1)
+} // Big O Notation for uniqueLinear: O(n)
+
+function uniqueQuadratic(words) {
+  const unique = [];                // 1 * 1 = O(1)
+  for (
+    let i = 0;                      // 1 * 1 = O(1)
+    i < words.length;               // 2 * n = O(n)
+    i++                             // 2 * n = O(n)
+  ) {
+    const word = words[i];          // 2 * n = O(n)
+    let isUnique = true;            // 1 * 1 = O(1)
+    for (
+      let c = 0;                    // 1 * n = O(n)
+      c < i;                        // 2 * n * n = O(n^2)
+      c++                           // 2 * n * n = O(n^2)
+    ) {
+      const comparing = words[c];   // 2 * n = O(n)
+      if (comparing === word) {     // 1 * n * n = O(n^2)
+        isUnique = false;           // 1 * n = O(1)
```

Comment:
Lines 34-36 are all executed n^2 times since they're inside the inner loop. Assume all code inside branches are executed each time when doing complexity analysis.